src/interpreter/executors/instruction/tests/reserved_registers.rs

```diff
@@ -0,0 +1,140 @@
+use super::*;
+use quickcheck::TestResult;
+use quickcheck_macros::quickcheck;
+
+// Ensure none of the opcodes can write to reserved registers
+#[quickcheck]
+fn cant_write_to_reserved_registers(opcode: u8, fuzz: u16) -> TestResult {
+    // opcode || REG_ONE || random fuzz
+    let attempt_reserved_reg_access = ((opcode as u32) << 24) | (1 << 18) | ((fuzz >> 4) as u32);
+    let instruction = Instruction::new(attempt_reserved_reg_access);
+    let opcode = Opcode::new(instruction);
+    // skip undefined opcodes
+    if matches!(opcode, Opcode::Undefined) {
+        return TestResult::discard();
+    }
+
+    let mut vm = Interpreter::with_memory_storage();
+
+    vm.init_script(CheckedTransaction::default())
+        .expect("Failed to init VM");
+    let res = vm.instruction(instruction);
+
+    if writes_to_ra(opcode) {
+        // if this opcode writes to $rA, expect an error since we're attempting to use a reserved register
+        if !matches!(
+            res,
+            Err(InterpreterError::PanicInstruction(r)) if r.reason() == &ReservedRegisterNotWritable
+        ) {
+            return TestResult::error(format!(
+                "expected ReservedRegisterNotWritable error {:?}",
+                (opcode, &res)
+            ));
+        }
+    } else {
+        // Ensure REG_ONE wasn't changed.
+        // While not a perfect guarantee against the opcode writing a value
+        // to an invalid register, this increases the likelihood of detecting
+        // erroneous register access.
+        if vm.registers[REG_ONE] != 1 {
+            return TestResult::error("reserved register was modified!");
+        }
```

Comment:
We should probably do this check irregardless of `writes_to_ra(opcode)`, as modifying REG_ONE to have any other value would be incorrect in either case.

Reply:
done